src/Microsoft.AspNetCore.Routing/CompositeEndpointDataSource.cs

```diff
@@ -106,5 +111,52 @@ private void CreateChangeToken()
             _cts = new CancellationTokenSource();
             _consumerChangeToken = new CancellationChangeToken(_cts.Token);
         }
+
+        private string DebuggerDisplayString => ToDebuggerDisplayString();
+
+        private string ToDebuggerDisplayString()
+        {
+            // Try using private variable '_endpoints' to avoid initialization
+            if (_endpoints == null)
+            {
+                return "No endpoints available yet to display.";
```

Comment:
nit: I think "No endpoints" is fine

Edit: Isn't _endpoints is assigned in the ctor and will never be null? Check for empty instead?

Reply:
> Edit: Isn't _endpoints is assigned in the ctor and will never be null? Check for empty instead?

No, it isn't assigned in the constructor. Its populated lazily.